src/C++/FileStore.cpp

```diff
@@ -31,10 +31,16 @@
 
 namespace FIX
 {
-FileStore::FileStore( std::string path, const SessionID& s )
-: m_msgFile( 0 ), m_headerFile( 0 ), m_seqNumsFile( 0 ), m_sessionFile( 0 )
+FileStore::FileStore( std::string path, const SessionID& s, bool deleteFile, FileManager* manager)
+: m_msgFile( 0 ), m_headerFile( 0 ), m_seqNumsFile( 0 ), m_sessionFile( 0 ), fileManager(manager)
 {
-  file_mkdir( path.c_str() );
+
+  if(!manager) {
+    fileManager = new FileManager();
+    defaultFileManagerUsed = true;
+  }
+
+  fileManager->makeDirectory( path.c_str() );
 
```

Comment:
Generated a FileManager nested Class to allow for stubbing out methods and injecting a test instance into unit tests to verify exceptions.  Without stubs producing the exceptions required for 100% line coverage was not possible.